geps/gep-1619.md

```diff
@@ -163,43 +163,258 @@ Java application servers such as Tomcat and Jetty, were the first to standardize
 
-To organize potential Gateway API configuration of cookie functionality, the next section will break down configuration into specific categories:
-- Session Cookie Configuration
-- Session Cookie Strategy
+To determine how the session persistence API should be designed, we must consider the following:
+1. Who is responsible for configuring it?
+2. What existing API object it is augmenting?
+3. What are the practical implications of attaching session persistence to an existing API object?
+4. How are existing implementation APIs designed?
 
-**Session Cookie Configuration**
+**Who is responsible for configuring it?**
 
-Session cookie configuration are the parameters and attributes the dataplane uses to create a cookie for establishing a session.
+Referring to the [Gateway API Security Model](https://gateway-api.sigs.k8s.io/concepts/security-model/#roles-and-personas),
+the target kubernetes role/persona for session persistence, are application developers. In most cases, session
+persistence is not considered an optional feature, but rather a necessary and deliberate component incorporated into the
+application's architecture by the developers themselves. Consequently, application developers should be responsible for
+adjusting the configuration to ensure the functionality of their applications.
 
-A set-cookie request can be broken down into [3 main components](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie):
-- Cookie Name
-- Cookie Value
-- Cookie Attributes
+**What existing API object it is augmenting?**
 
-Only the Cookie Name and Cookie Value are required. All Cookie Attributes are optional. Attributes are simply key=value pairs with the value as optional for some attributes.
+Session persistence augments a Kubernetes service object after the routing to the service has taken place. In
+Kubernetes, a service is an abstract representation of a set of pods that work together to provide a specific
+functionality. Services perform load balancing, routing traffic from a service endpoint to a pod. Session persistence
+augments the load balancing functionality of the service by consistently directing traffic to the same pod.
 
-The set-cookies attributes defined by [rfc6265](https://www.rfc-editor.org/rfc/rfc6265#section-5.2) are:
+**What are the practical implications of attaching session persistence to an existing API object?**
+
+One significant practical implication of adding session persistence to services is its reusability across multiple
+routes. This reusability can be advantageous as it eliminates the need for repetitive session persistence configuration
+for each individual route. However, on the flip side, it also mandates the inclusion of session persistence
+configuration for every route that utilizes the service. If session persistence is a requirement of the application
+architecture, there is no need to configure it separately for each route.
```

Comment:
On the flip side when weight-based routing is in place where we split the traffic and send it to different services (based on specified percentages) we want to make sure that session persistence works across all the services. So the cookie will direct a request to a service *and* a pod (backend) within that service.

Reply:
Agree, updated.